service/frontend/admin_handler.go

```diff
@@ -1884,6 +1885,13 @@ func (adh *AdminHandler) StreamWorkflowReplicationMessages(
 			resp, err := serverCluster.Recv()
 			if err != nil {
 				logger.Info("AdminStreamReplicationMessages server -> client encountered error", tag.Error(err))
+				var solErr *serviceerrors.ShardOwnershipLost
+				var suErr *serviceerror.Unavailable
+				if errors.As(err, &solErr) || errors.As(err, &suErr) {
+					ctx, cl := context.WithTimeout(context.Background(), 2*time.Second)
+					adh.historyClient.DescribeHistoryHost(ctx, &historyservice.DescribeHistoryHostRequest{ShardId: serverClusterShardID.ShardID})
```

Comment:
curious why this is not working. https://github.com/temporalio/temporal/pull/6482

Reply:
Because `client.StreamWorkflowReplicationMessages` just to establish the stream connection. Data exchange happened when Recv() is called, and that is the time we got ShardOwnershipLost error.